File: 프로그래머스/unrated/155652. 둘만의 암호/둘만의 암호.py
# a = 97
# z = 122
def solution(s, skip, index):
    new = ''
    for S in s:
        count = change = 0
        while count != index:
            change += 1
            if chr((((ord(S)+change)-97) % 26) + 97) not in skip:count += 1             
        new += chr((((ord(S)+change)-97) % 26) + 97)
    print(ord('a'))
    print(ord('z'))
    print(ord('A'))
    print(ord('Z'))
    print(chr(97))
    
    return new

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.js
function solution(s, skip, index) {
    result = ""
    Array.from(s).forEach((alpha) => {
        let chage = 1, pass = 0
        while (chage <= index){
            if (Array.from(skip).includes(String.fromCharCode((alpha.charCodeAt()+chage+pass-97) % 26 + 97))){
                pass++
            }
            else {
                chage ++
            }
        }
        result += String.fromCharCode(((alpha.charCodeAt() + chage + pass - 98) % 26) + 97)
    })    
    return result;
}  